### DISCUSSION TITLE
discharge.nc not working in my Mac Air M1.

### DISCUSSION THREAD

**Discussion by prabhatbanjara** [mhm-ufz/mHM#49] [Q&A] [answered]
Hello team, This is me Prabhat Banjara working on mhm from Nepal. I am finding corrupt file only on discharge.nc other .nc files on output can be opened using Panoply. How can i solve this problem?

**Comment by pallavhydro** [comment]
@prabhatbanjara 

Interesting problem. Could you perhaps share some screenshots of the error you get (e.g. with a `cdo sinfon`)
 on all three .nc files of mhm output?

**Reply by prabhatbanjara** [reply]
<img width="912" alt="Screen Shot 2022-02-10 at 11 44 22 PM" src="https://user-images.githubusercontent.com/99330118/153468244-78115285-0cdb-4943-866c-b57cff34a129.png">

Hlo sir i got the problem in dicharge.nc file in the output while other .nc file were easily opened using panoply. While opening discharge,nc file there was remark like Dataobject doesnt start with OHDR

**Comment by prabhatbanjara** [comment]
<img width="563" alt="Screen Shot 2022-02-08 at 5 19 41 PM" src="https://user-images.githubusercontent.com/99330118/153468651-3fa2e5dd-bc6b-494f-b963-4e1444348cf5.png">

**Comment by pallavhydro** [comment]
Interesting that the -Fluxes_States.nc files behave properly but somehow the discharge.nc has odd behaviour. There was a similar case in #46 where the OS was Windows 11.

@MuellerSeb there are two identical issues now from two different OS. Would it make sense to check (or improve) how discharge.nc is stored in the source code? Or could this be attributed to particular version of the compiler?

**Reply by StephanThober** [reply]
Hi @pallavhydro ,

yes, the issue seems to be specific to discharge.nc.

@prabhatbanjara : could you provide the discharge.nc here so we could double check? Could you also try to open the file with ncview and share the error message, that you get.

Thanks,
Stephan

**Comment by prabhatbanjara** [comment]
[image: Screen Shot 2022-03-03 at 5.21.01 PM.png]

On Thu, 3 Mar 2022 at 15:48, StephanThober ***@***.***> wrote:

> Hi @pallavhydro <https://github.com/pallavhydro> ,
>
> yes, the issue seems to be specific to discharge.nc.
>
> @prabhatbanjara <https://github.com/prabhatbanjara> : could you provide
> the discharge.nc here so we could double check? Could you also try to
> open the file with ncview and share the error message, that you get.
>
> Thanks,
> Stephan
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mhm-ufz/mhm/discussions/49#discussioncomment-2286607>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXV2QRSDFQ3CAER5LL3LGZLU6CE65ANCNFSM5N5GCZZQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
With Best Regards
*Er. Prabhat Banjara*
9863033194
Water Resource Engineer

**Comment by prabhatbanjara** [comment]
hlo sir, I am sharing both discharge.nc file and screenshot of the error
message here sir

On Thu, 3 Mar 2022 at 15:48, StephanThober ***@***.***> wrote:

> Hi @pallavhydro <https://github.com/pallavhydro> ,
>
> yes, the issue seems to be specific to discharge.nc.
>
> @prabhatbanjara <https://github.com/prabhatbanjara> : could you provide
> the discharge.nc here so we could double check? Could you also try to
> open the file with ncview and share the error message, that you get.
>
> Thanks,
> Stephan
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mhm-ufz/mhm/discussions/49#discussioncomment-2286607>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXV2QRSDFQ3CAER5LL3LGZLU6CE65ANCNFSM5N5GCZZQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
[image: Screen Shot 2022-03-03 at 5.21.01 PM.png]

-- 
With Best Regards
*Er. Prabhat Banjara*
9863033194
Water Resource Engineer

**Comment by MuellerSeb** [comment]
Unfortunately I dont see the image and also not the file. Could you append these files to your message so we can download them?

**Comment by PrabhatBanjaraa** [comment]
Hlo sir, I am not able to upload .nc file here how can i share it as when i try to drag and drop .nc file this message appeared sir.

![](https://files.slack.com/files-pri/T025AUHG3PY-F0352056U4X/screen_shot_2022-03-03_at_6.37.17_pm.png)

**Reply by PrabhatBanjaraa** [reply]
[discharge.nc.zip](https://github.com/mhm-ufz/mhm/files/8178023/discharge.nc.zip)
Here is discharge.nc file sir.

**Reply by PrabhatBanjaraa** [reply]
<img width="545" alt="Screen Shot 2022-03-03 at 5 21 01 PM" src="https://user-images.githubusercontent.com/100848461/156571297-b20b2d0c-85b2-4601-9454-e4b580ab74ca.png">
Here is the screenshot of the error message when i tried to open discharge.nc file in panoply other mHM_Fluxes_States.nc and mRM_Fluxes_States.nc were opened in panoply sir.

**Comment by StephanThober** [comment]
I confirm that the file is broken. Apparently, the time dimension was created, but something went wrong in the creation of the variable. Could you share with us your mhm.nml and the log file?


<img width="209" alt="Screenshot 2022-03-03 at 14 56 58" src="https://user-images.githubusercontent.com/19333234/156578929-2bc2ed1e-4aa1-4691-9c6c-01f7b6eee4c6.png">

**Reply by PrabhatBanjaraa** [reply]
[ConfigFile.log](https://github.com/mhm-ufz/mhm/files/8178424/ConfigFile.log)
[mhm.nml.zip](https://github.com/mhm-ufz/mhm/files/8178425/mhm.nml.zip)

**Reply by StephanThober** [reply]
Thanks for sharing this, this looks fine. Could you activate the writing of the discharge field in the mrm_outputs.nml and share the mRM_Fluxes_States.nc  that would then be written?

**Comment by PrabhatBanjaraa** [comment]
[mrm_outputs.nml.zip](https://github.com/mhm-ufz/mhm/files/8190977/mrm_outputs.nml.zip)
[mHM_Fluxes_States.nc.zip](https://github.com/mhm-ufz/mhm/files/8190978/mHM_Fluxes_States.nc.zip)
Here in mrm_outputs.nml it is true and the output of it in output folder is shared here sir.

**Reply by StephanThober** [reply]
Thank you, was there also a mRM_Fluxes_States.nc file written? It should be there.

**Reply by PrabhatBanjaraa** [reply]
Both mHM_Fluxes_States.nc and mRM_Fluxes_States.nc could be opened using panoply but the problem was only with discharge.nc file sir

**Comment by PrabhatBanjaraa** [comment]
[mRM_Fluxes_States.nc.zip](https://github.com/mhm-ufz/mhm/files/8202873/mRM_Fluxes_States.nc.zip)
This is mRM_Fluxes_States.nc file sir this file is opened using panoply sir

**Comment by StephanThober** [comment, marked as answer]
@prabhatbanjara : sorry for the slow replies here. I still would like to understand what is going on. The way this netcdf file is written is actually different than the other netcdf files and using sort of outdated routines. This might explain why this file is not written correctly while the other are. Additionally, this file is sort of redundant because there is the text file daily_discharge.out that contains the same information. Could you please let me know whether this text file is written correctly. Thanks.=

**Reply by PrabhatBanjaraa** [reply]
Hlo sir the discharge.out file is written correctly we can open it and it contains observed and simulated discharge.

**Reply by StephanThober** [reply]
For the time being, I would suggest you to work with the text file. We will need to update the output writing routine, but this might take some time. I created an issue for this in our git repo. Please note that other users reported that the problem was solved by reinstalling mHM #46 . I hope this is fine for you.

**Comment by prabhatbanjara** [comment]
Hlo mHM team. I am working in mHM model from Nepal and i got the problem
which I am sharing here in screenshot below. It is stating mHM generated x
and y are not matching. How can i deal with this problem?
 [image: Screen Shot 2022-05-23 at 9.34.37 PM.png]

On Tue, 22 Mar 2022 at 16:00, StephanThober ***@***.***>
wrote:

> For the time being, I would suggest you to work with the text file. We
> will need to update the output writing routine, but this might take some
> time. I created an issue for this in our git repo. Please note that other
> users reported that the problem was solved by reinstalling mHM #46
> <https://github.com/mhm-ufz/mhm/discussions/46> . I hope this is fine for
> you.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mhm-ufz/mhm/discussions/49#discussioncomment-2413499>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXV2QRUZDUXYXNMBD3WMRH3VBGMSRANCNFSM5N5GCZZQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
With Best Regards
*Er. Prabhat Banjara*
9863033194
Water Resource Engineer

**Reply by StephanThober** [reply]
Dear @prabhatbanjara , I cannot see the screenshot. The error message is not that informative, but it occurs when the x and y dimensions of a variable that is read from a netcdf file is not matching what mHM expects based on the dem and the provide hydrology_resolution. It can occur for different input files and it should be visible from the stdout which file it is. Could you please share the stdout of mHM?

**Reply by PrabhatBanjaraa** [reply]
<img width="633" alt="Screen Shot 2022-05-23 at 9 34 37 PM" src="https://user-images.githubusercontent.com/100848461/169982535-84a63c3e-9157-444e-ab62-42d72a9df55d.png">

**Reply by StephanThober** [reply]
@prabhatbanjara ok, so the issues is with the precipitation input file. You need to check 

1. the number of rows and columns in your dem file
2. the x and y dimensions in your netcdf file, the correct ordering of the dimensions in the netcdf file (it should be time, y, x; see also meteo input in test_domain folder)
3. your hydrology resolution setting

My guess would be that the dimensions in the netcdf file are not correct, so point 2.

**Reply by pallavhydro** [reply]
@prabhatbanjara 

Could you please share 

1. `header.txt` file that you are using in your `meteo folder` here
2. And also the screenshot of `cdo sinfon pre.nc | head`

We need to check whether these two are saying same or different things to mHM.

**Comment by prabhatbanjara** [comment]
For_dudhkoshi_after_mask_noerror _after_final_p...
<https://drive.google.com/file/d/1gLfuyQ95XY0dJDF4aj1wNRkr26UGZIkd/view?usp=drive_web>
This is the setup for Nepal basin and i wanted to run mhm for this
setup and the problem was as stated earlier.

On Tue, 24 May 2022 at 13:05, StephanThober ***@***.***>
wrote:

> Dear @prabhatbanjara <https://github.com/prabhatbanjara> , I cannot see
> the screenshot. The error message is not that informative, but it occurs
> when the x and y dimensions of a variable that is read from a netcdf file
> is not matching what mHM expects based on the dem and the provide
> hydrology_resolution. It can occur for different input files and it should
> be visible from the stdout which file it is. Could you please share the
> stdout of mHM?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mhm-ufz/mhm/discussions/49#discussioncomment-2809718>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXV2QRW6IHA2ZL2TAK53V7DVLR7NTANCNFSM5N5GCZZQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
With Best Regards
*Er. Prabhat Banjara*
9863033194
Water Resource Engineer

**Comment by prabhatbanjara** [comment]
For_dudhkoshi_after_mask_noerror _after_final_p...
<https://drive.google.com/file/d/1mDAA3R_JRAJ73DayRXn197DKjGA6V12q/view?usp=drive_web>
sir here is the whole setup for a basin Dudhkoshi having control input all
files

On Tue, 24 May 2022 at 15:53, Pallav Kumar Shrestha <
***@***.***> wrote:

> @prabhatbanjara <https://github.com/prabhatbanjara>
>
> Could you please share
>
>    1. header.txt file that you are using in your meter folder here
>    2. And also the screenshot of cdo sinfon pre.nc | head
>
> We need to check whether these two are saying same or different things to
> mHM.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mhm-ufz/mhm/discussions/49#discussioncomment-2810897>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXV2QRSR5SKDRGFNXAUU3V3VLSTAZANCNFSM5N5GCZZQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
With Best Regards
*Er. Prabhat Banjara*
9863033194
Water Resource Engineer

**Comment by pallavhydro** [comment]
If I am not wrong, your `meteo/header.txt` and coordinates of `meteo/pre.nc` look like this.

<img width="215" alt="Screenshot 2022-05-24 at 12 51 05" src="https://user-images.githubusercontent.com/58865333/170018216-c68c343e-ea16-4125-a3a8-a47862ae8283.png">

<img width="725" alt="Screenshot 2022-05-24 at 12 52 10" src="https://user-images.githubusercontent.com/58865333/170018440-40f6c4f8-aeb2-4e69-8642-265870671c91.png">

As per @StephanThober 's  guess, your header.txt (256 x 224) is not matching meteo netcdf files (16 x 14). Also, the cellsize is not correct.

Please update the header.txt as follows -
```
ncols 16
nrows 14
xllcorner    85.000000000000
yllcorner    26.000000000000
cellsize     0.25
NODATA_value -9999
```

Let us know if it works for you.

**Reply by PrabhatBanjaraa** [reply]
sir header.txt can be updated manually or we have to do this by anyother tool?

**Reply by pallavhydro** [reply]
The header files can be edited manually, like namelist (.nml) files.

**Reply by PrabhatBanjaraa** [reply]
<img width="676" alt="Screen Shot 2022-05-24 at 4 46 07 PM" src="https://user-images.githubusercontent.com/100848461/170020003-ffc06b32-6024-4f71-a314-a39b4c7be0a5.png">
sir this is new header.txt as provided by olda so i need to make ncols=16, nrows=14, and cellsize=0.25?

**Reply by StephanThober** [reply]
yes, please try and let us know.

**Reply by PrabhatBanjaraa** [reply]
It worked sir. Thank you mHM team sir

**Comment by MuellerSeb** [comment]
Will be solved in the next release:
https://git.ufz.de/mhm/mhm/-/merge_requests/135

**Comment by pallavhydro** [comment]
This has been solved in the latest release of mHM.

Re-install of mHM should solve it :-)
